mod-source-record-storage-server/src/main/java/org/folio/services/RecordServiceImpl.java

```
@@ -300,11 +313,63 @@ public Future<Void> updateRecordsState(String matchedId, RecordState state, Reco
     return recordDao.updateRecordsState(matchedId, state, recordType, tenantId);
   }
 
+  @Override
+  public Future<RecordsIdentifiersCollection> getMatchedRecordsIdentifiers(RecordMatchingDto recordMatchingDto, String tenantId) {
+    MatchField matchField = prepareMatchField(recordMatchingDto);
+    TypeConnection typeConnection = getTypeConnection(recordMatchingDto.getRecordType());
+
+    if (matchField.isDefaultField()) {
+      return processDefaultMatchField(matchField, tenantId, typeConnection,
+        recordMatchingDto.getOffset(), recordMatchingDto.getLimit());
+    }
+    return recordDao.getMatchedRecordsIdentifiers(matchField, typeConnection, true,
+      recordMatchingDto.getOffset(), recordMatchingDto.getLimit(), tenantId);
+  }
+
+  private MatchField prepareMatchField(RecordMatchingDto recordMatchingDto) {
+    // only one matching filter is expected in the current implementation for processing records matching
+    Filter filter = recordMatchingDto.getFilters().get(0);
```

We should throw an exception here, if the number of filters is greater than 1.